### PR TITLE
Doc (registry): Update to the image registry doc

### PR DESCRIPTION
**Pull request by hickeyma**
While using the "Configure image registry" doc, I noticed some improvements which could help users. This PR adds these improvements.

### CODE REVIEW

File: docs/registry.md [hunk 1]
```diff
@@ -58,7 +64,7 @@ version = 2
     key_file  = "key.pem"
 ```
 
-In the config example shown above, TLS mutual authentication will be used for communications with the registry endpoint located at https://my.custom.registry.
+In the config example shown above, TLS mutual authentication will be used for communications with the registry endpoint located at <https://my.custom.registry>.
```
Comment (mikebrow) [inline docs/registry.md:67]: did you want a `[ref](link)`

File: docs/registry.md [hunk 2]
```diff
@@ -98,40 +105,51 @@ version = 2
...
 
 Please note that auth config passed by CRI takes precedence over this config.
 The registry credential in this config will only be used when auth config is
 not specified by Kubernetes via CRI.
 
-After modify this config, you need restart the `containerd` service.
+After modifying this config, you need to restart the `containerd` service.
 
-### Configure Registry Credentials Example - GCR with _json_key Authentication
+### Configure Registry Credentials Example - GCR with service account key authentication
```
Comment (mikebrow) [inline docs/registry.md:117]: title caps

File: docs/registry.md [hunk 3]
```diff
@@ -98,40 +105,51 @@ version = 2
...
+* Enable GCR for your project (see [Quickstart for Container Registry](https://cloud.google.com/container-registry/docs/quickstart))
+* For authentication to GCR: Create [service account and JSON key](https://cloud.google.com/container-registry/docs/advanced-authentication#json-key)
+* The JSON key file needs to be downloaded to your system from the GCP console
+* For access to the GCR storage: Add service account to the GCR storage bucket with storage admin access rights (see [Granting permissions](https://cloud.google.com/container-registry/docs/access-control#grant-bucket))
+
+Refer to [Pushing and pulling images](https://cloud.google.com/container-registry/docs/pushing-and-pulling) for detailed information on the above steps.
 
-Generate a single line for the _json_key file that you downloaded:
+> Note: The JSON key file is a multi-line file and can be cumbersome to use its contents as a key outside of file. It is worthwhile generating a single line format output of the file. One way of doing this is using the `jq` tool as follows: `jq -c . key.json`
```
Comment (mikebrow) [inline docs/registry.md:129]: s/and can/and it can/
s/its/the contents/
s/of file/of the file/

**Review by mikebrow — COMMENTED** [review]
looking good just a couple nits

File: docs/registry.md (hunk 1 above)
Comment (hickeyma) [reply docs/registry.md:67]: Markdown linter prefers it within `<>`, thats all

File: docs/registry.md (hunk 2 above)
Comment (hickeyma) [reply docs/registry.md]: Updated

File: docs/registry.md (hunk 3 above)
Comment (hickeyma) [reply docs/registry.md]: Updated

**Comment by hickeyma** [comment]
Thanks for the review @mikebrow. Updated and ready for review again.

File: docs/registry.md (hunk 1 above)
Comment (hickeyma) [reply docs/registry.md:67]: What do you think?

**Review by mikebrow — APPROVED** [review]
LGTM

File: docs/registry.md (hunk 1 above)
Comment (mikebrow) [reply docs/registry.md:67]: unfortunately the preview code on github here... would not show what that ^ looks like.  After merging.. I see it shows it as a clickable link.  I don't think that was the intent. Maybe `https://my.custom.registry.` Cheers!

**Pull request merged**